### ISSUE TITLE
Support of an extensible list of different L1. Tx crafter.

### ISSUE THREAD

**Issue by mm-s** [openwallet-foundation/architecture-sig#26]
One of the most clear example of importing features from other codebases is the 'transaction crafter' module. The different settlement networks (L1) (Bitcoin, ..) provide the spec and implementation and I think the OWF should be extensible (e.g. with plugins) implementing support for the different L1.
This means there is a need to define a spec or interface for interacting with arbitrary L1 systems.

**Comment by OR13** [comment]
Yes, there are a few cases where a wallet needs to craft a TX, and then might submit that TX to a network.

censorship / surveillance issues usually arise depending on which network they attempt to submit the TX to, and the timings of that event in relation to other activity.

typical interface is:


(craft tx) -> sign (in wallet) -> submit to network


crafting could be internal to the wallet, or external.

Examples of internal tx crafting include NFT transfer operations.

**Comment by adrianhopebailie** [comment]
This should also consider tx that get submitted to non-blockchain networks like card/bank networks.

Consider the work at W3C on [Secure Payment Confirmation](https://www.w3.org/TR/secure-payment-confirmation/) (based on FIDO) where the user provides a cryptographically signed assertion that they wish to complete a transaction.
